src/Data/Functor/Foldable/TH.hs

```diff
@@ -171,17 +205,27 @@ instance MakeBaseFunctor Dec where
 
 -- | Rules of renaming data names
 data BaseRules = BaseRules
-    { _baseRulesType  :: Name -> Name
-    , _baseRulesCon   :: Name -> Name
-    , _baseRulesField :: Name -> Name
+    { _baseRulesType               :: Name -> Name
+    , _baseRulesCon                :: Name -> Name
+    , _baseRulesField              :: Name -> Name
+    , _baseRulesIncludeBase        :: Bool
+    , _baseRulesIncludeRecursive   :: Bool
+    , _baseRulesIncludeCorecursive :: Bool
```

Comment:
There are likely some unrepresentable states here. For instance, I don't think it ever makes sense to generate a `Recursive` or `Corecursive` instance without a corresponding `Base` instance. It might make more sense to combine these into a single field. Something like:

```hs
data BaseRules = BaseRules
  { ...
  , _baseRulesApproach :: BaseRulesApproach
  }

data BaseRulesApproach
  = RecursiveOnly
  | CorecursiveOnly
  | RecursiveAndCorecursive
```

Reply:
Actually, I originally was using this _just_ for the Base data type, no classes, so I suppose it makes sense to consider generating an instance of just the Base data type but not the classes.

Reply:
> I don't think it ever makes sense to generate a `Recursive` or `Corecursive` instance without a corresponding `Base` instance

I do think it can make sense. yaya likes to use base functors which are independent of their recursive type: for example, its base functor for lists is called `XNor`, not `ListF`.

```haskell
data XNor a b = Neither | Both a b
  deriving Functor
```

This `XNor` is clearly intended to be independently useful. It would thus make sense to import `XNor` from a separate library, and to generate `Recursive` and `Corecursive` instances which use that existing type which use this existing `XNor` instead of generating a new type:

```
data List a = Nil | Cons a (List a)

makeBaseFunctorWith ( baseRules
                    & baseRulesIncludeBase .~ False
                    & baseRulesType .~ \case
                        Name (OccName "List") nameFlavor -> Name (OccName "XNor") nameFlavor
                    & baseRulesCon  .~ \case
                        Name (OccName "Nil") nameFlavor -> Name (OccName "Neither") nameFlavor
                        Name (OccName "Cons") nameFlavor -> Name (OccName "Both") nameFlavor
                    )
                    ''List
```